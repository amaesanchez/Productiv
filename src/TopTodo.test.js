import { render, screen } from '@testing-library/react';
import TopTodo from "./TopTodo"

const TODOS_TEST = [
  {
    id: 1,
    title: "Code!",
    description: "Write some code",
    priority: 2,
  },
  {
    id: 2,
    title: "Make dinner",
    description: "Cook something healthy",
    priority: 1,
  },
  {
    id: 3,
    title: "Go to bed",
    description: "In bed by 11:15",
    priority: 3,
  },
];

describe("render without crashing", function () {
  it("renders without crashing", function() {
    render(<TopTodo todos={ TODOS_TEST } />)
  });
});

describe("snapshot tests for stability", function () {
  it("matches snapshot", function () {
    const { container } = render(<TopTodo todos={ TODOS_TEST } />);
    expect(container).toMatchSnapshot();
  });
});

it("Renders appropriate todo item in Top Todo area of DOM", function () {
  const { queryByText } = render(<TopTodo todos={ TODOS_TEST } />);

  const todo = queryByText("Cook something healthy")

  expect(todo).toBeInTheDocument();
})
//getAllByText
