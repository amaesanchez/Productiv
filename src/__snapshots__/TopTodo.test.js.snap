// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests for stability matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        Make dinner
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        Cook something healthy
      </small>
    </div>
  </div>
</div>
`;
